#!/usr/bin/env python3
PACKAGE = "copilot_interface"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("depthHoldToggle",   bool_t,   0, "Enable depth hold",  False)
gen.add("targetDepth",   double_t,   0, "Choose depth to be held", 7.5, 0, 15)
gen.add("pScalar",    double_t,    0, "PID: proportional", 1,  0, 2)
gen.add("iScalar",    double_t,    0, "PID: integral", .1,  0, .5)
gen.add("dScalar",    double_t,    0, "PID: derivative", .05,  0, .5)
gen.add("autoDock",   bool_t,   0, "Enable Autonomous Docking", False)
exit(gen.generate(PACKAGE, "depthHoldControls", "depthControlParams"))
